backend/drive/drive.go

```diff
@@ -870,6 +870,11 @@ func (f *Fs) getFile(ctx context.Context, ID string, fields googleapi.Field) (in
 	return info, err
 }
 
+// Options returns remote-specific options for reference by commands
```

Comment:
I don't like this - this breaks the backend encapsulation and will mean you need to `import "github.com/rclone/rclone/backend/drive` to use it which means that you ignore the users backend choices if they are embedding rclone.

The proper way to do this is to add a new method or a new Feature flag.

However I think you can do something different.

Detect whether you ever receive a file with size < 0 before giving your warning with a `sync.Once`.

This will mean that you catch the other cases because files of unknown length can happen in other backends not just Google Drive

- google drive
- google photos
- s3/gcs with Content-encoding Gzip files

I think this PR should be called `bisync: add support files with unknown length, including Google Docs - fixes #5696`

Generally in this commit `s/Google doc/files of unknown size/g;` :-)

Reply:
The main reason for not using size is that Directories also have a size of -1 (if using `--create-empty-src-dirs`). But we do have another way of detecting Dirs, so I think actually I can make this work. 

I had also wanted to not bother users with this warning if they're already using `--drive-skip-gdocs` (which we can't otherwise check). But I think you are right that we can assume they are not, if we see non-dir listings with size of -1.

Reply:
Made this change.